lib/registry/src/graphql.rs

```diff
@@ -0,0 +1,251 @@
+        } else {
+            return Ok(None);
+        }
+        .map_err(|e| ProxyError::ConnectionError(e.to_string()))
+        .and_then(
+            |(proxy_url_str, proxy, url_location): (String, _, &'static str)| {
+                url::Url::parse(&proxy_url_str)
+                    .map_err(|e| ProxyError::UrlParseError {
+                        url_location: url_location.to_string(),
+                        error_message: e.to_string(),
+                    })
+                    .map(|url| {
+                        if !(url.username().is_empty()) && url.password().is_some() {
+                            proxy.basic_auth(url.username(), url.password().unwrap_or_default())
+                        } else {
+                            proxy
+                        }
+                    })
+            },
+        )?;
+
+        Ok(Some(proxy))
+    }
+}
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/schema.graphql",
+    query_path = "graphql/queries/get_package_version.graphql",
+    response_derives = "Debug"
+)]
+pub(crate) struct GetPackageVersionQuery;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/schema.graphql",
+    query_path = "graphql/queries/get_package_by_command.graphql",
+    response_derives = "Debug"
+)]
+pub(crate) struct GetPackageByCommandQuery;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/schema.graphql",
+    query_path = "graphql/queries/test_if_registry_present.graphql",
+    response_derives = "Debug"
+)]
+pub(crate) struct TestIfRegistryPresent;
+
+#[cfg(target_os = "wasi")]
+pub fn whoami_distro() -> String {
+    whoami::os().to_lowercase()
+}
+
+#[cfg(not(target_os = "wasi"))]
+pub fn whoami_distro() -> String {
+    whoami::distro().to_lowercase()
+}
+
+pub fn execute_query_modifier_inner_check_json<V, F>(
+    registry_url: &str,
+    login_token: &str,
+    query: &QueryBody<V>,
+    timeout: Option<Duration>,
+    form_modifier: F,
+) -> anyhow::Result<()>
+where
+    V: serde::Serialize,
+    F: FnOnce(Form) -> Form,
+{
+    let client = {
+        let builder = Client::builder();
+
+        let builder = if let Some(proxy) = proxy::maybe_set_up_proxy()? {
+            builder.proxy(proxy)
+        } else {
+            builder
+        };
+        builder.build()?
+    };
+
+    let vars = serde_json::to_string(&query.variables).unwrap();
+
+    let form = Form::new()
+        .text("query", query.query.to_string())
+        .text("operationName", query.operation_name.to_string())
+        .text("variables", vars);
+
+    let form = form_modifier(form);
+
+    let user_agent = format!(
+        "wapm/{} {} {}",
+        env!("CARGO_PKG_VERSION"),
+        whoami::platform(),
+        whoami_distro(),
+    );
+
+    let mut res = client
+        .post(registry_url)
+        .multipart(form)
+        .bearer_auth(env::var("WAPM_REGISTRY_TOKEN").unwrap_or_else(|_| login_token.to_string()))
```

Comment:
Let's use WASMER_TOKEN and fall back to WAPM_REGISTRY_TOKEN and then the login_token